src/fudge/log.clj

```diff
@@ -52,12 +61,15 @@
       {:message data}))
 
 (defn prepare-data-for-logging
-  "Prepare the `data` structure being logged by setting the `:level`
-   and `:date` values"
-  [config data level]
+  "Prepare the `data` structure being logged by setting the `:level`,
+   `:date`, `:ns`, `:file` and `:line` values"
+  [config data level calling-meta]
   (->> data
        normalize-data
        (merge {:date (call config :date-fn)
+               :ns (:ns calling-meta)
+               :file (:file calling-meta)
+               :line (:line calling-meta)
```

Comment:
Yes, good point r.e. the merge.

I was actually thinking that maybe `data` should be the sub key. That's the user provided bit, so that's the bit which might have clashing keys depending on what the user wants to output. `data` in a subkey stops anything like that.